src/xopen/__init__.py

```diff
@@ -508,15 +508,26 @@ def _open_zst(
     assert compresslevel != 0
     if compresslevel is None:
         compresslevel = XOPEN_DEFAULT_ZST_COMPRESSION
+    if zstandard:
+        max_window_bits = zstandard.WINDOWLOG_MAX
+    else:
+        max_window_bits = 31
     if threads != 0:
         try:
             # zstd can compress using multiple cores
+            program_args: Tuple[str, ...] = ("zstd",)
+            if "r" in mode:
+                # Only use --long=31 for decompression. Otherwise compression
+                # will always use a very long window size, which will result
+                # in very long compute times and the resulting archive not
```

Comment:
As I understand it (and my tests confirmed), a large window size doesn’t result in long compute times at compression time, only high memory usage

Reply:
Considering a window size that does not fit in the CPU L1 or L2 cache will lead to very slow compute times. But the more important reason not to do it is that all the zstd files produced by xopen cannot be opened by zstd unless the user fiddles with the settings. That is very user unfriendly. Also --long=31 is very much unnessecary for most use cases.

So I will adjust the comment a bit.